### PR TITLE
:sparkles: Repurpose report post modal and re-use for list reporting

### PR DESCRIPTION
**Pull request by foysalit**
~I am not sure if the contextual options (spam, copyright violation etc.) all applies to both list and post but generally speaking, I think having a default/fallback modal for reporting any kind of content could be beneficial.~

This PR refactors and generalizes the reporting modal to make it easier to report any record in the network. As an initial offering, using the same component, we are reporting lists and feed along with previously available account and post reporting.

### TODO
- [x] Write tests
- [x] Fix types

https://github.com/bluesky-social/social-app/assets/1919066/92315f35-4a6c-434b-bba0-2c897af9deca

### CODE REVIEW

**Review by pfrazee — CHANGES_REQUESTED** [review]
If we want to repurpose the `ReportPostModal`, we should make its semantics generic (`ReportContent`). The parameters should then be changed to `uri` and `cid`.

You can intuit what kind of content it is by looking at the `uri` by parsing with the `AtUri` library (from the api) and then looking at the `parsedUri.collection` property, dropping the need for a `collection` parameter. From there you can tune the copy and the reporting options based on its value.

**Comment by foysalit** [comment]
@pfrazee  Agreed! it was kinda hastily put together as a PoC and check if we want to take this direction. If you're onboard, I will make a pass through the semantics and may be even unify report account thing under this one report-content modal.

**Comment by pfrazee** [comment]
SGTM!

File: .eslintrc.js [hunk 1]
```diff
@@ -16,6 +16,8 @@ module.exports = {
     '*.lock',
     '.husky',
     'patches',
+    'bskyweb',
```
Comment (foysalit) [inline .eslintrc.js:19]: was getting lint error trying to commit after conflict resolution.

File: src/view/com/modals/report/Modal.tsx [hunk 2]
```diff
@@ -79,141 +107,73 @@ export function Component({
...
-        What is the issue with this post?
+        What is the issue with this {collectionName}?
       </Text>
-      <RadioGroup
-        testID="reportPostRadios"
-        items={ITEMS}
-        onSelect={onSelectIssue}
+      <ReportReasonOptions
+        atUri={atUri}
+        selectedIssue={issue}
```
Comment (foysalit) [inline src/view/com/modals/report/Modal.tsx:167]: Without this, when you go to details input field and come back to the list, the previously selected reason disappears even though, in state, it still has that reason selected, just not showing in the UI.

File: src/view/index.ts [hunk 3]
```diff
@@ -181,5 +182,6 @@ export function setup() {
     faXmark,
     faPlay,
     faPause,
+    faList,
```
Comment (foysalit) [inline src/view/index.ts:185]: This fixes list icon rendering on web under dropdown menus

File: __e2e__/tests/mute-lists.test.ts [hunk 4]
```diff
@@ -114,7 +114,7 @@ describe('Mute lists', () => {
 
   it('Shows the mutelist on my profile', async () => {
     await element(by.id('bottomBarProfileBtn')).tap()
-    await element(by.id('selector-2')).tap()
+    await element(by.id('selector-3')).tap()
```
Comment (foysalit) [inline __e2e__/tests/mute-lists.test.ts:117]: I am now seeing the `Media` tab on `selector-2` so updating to 3 here which makes the test pass.

File: src/view/com/modals/report/ReasonOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import React, {useMemo} from 'react'
+import {AtUri, ComAtprotoModerationDefs} from '@atproto/api'
+
+import {Text} from '../../util/text/Text'
+import {UsePaletteValue, usePalette} from 'lib/hooks/usePalette'
+import {RadioGroup, RadioGroupItem} from 'view/com/util/forms/RadioGroup'
+import {CollectionId} from './types'
+
+type ReasonMap = Record<string, {title: string; description: string}>
+const CommonReasons = {
```
Comment (foysalit) [inline src/view/com/modals/report/ReasonOptions.tsx:11]: Primary goal with this refactor is to kinda split data from the UI. Going forward, adding new reasons to existing or new collections should be relatively easier with this model.

**Review by pfrazee — APPROVED** [review]
🚀 Good update! Thanks for factoring in the changes I requested

**Pull request merged**